x-pack/metricbeat/module/tomcat/cache/_meta/data.json

```diff
@@ -0,0 +1,67 @@
+{
+    "@timestamp": "2019-09-19T13:42:47.806Z",
+    "@metadata": {
+        "beat": "metricbeat",
+        "type": "_doc",
+        "version": "8.0.0"
+    },
+    "event": {
+        "dataset": "tomcat.cache",
+        "module": "tomcat",
+        "duration": 98441554
+    },
+    "metricset": {
+        "name": "cache",
+        "period": 10000
+    },
+    "service": {
+        "address": "localhost:8080",
+        "type": "tomcat"
+    },
+    "tomcat": {
+        "cache": {
+            "mbean": "Catalina:context=/docs,host=localhost,name=Cache,type=WebResourceRoot",
```

Comment:
@jsoriano We are looking into why the **tomcat** module is not working for us, and the issue is that the domain should be `Tomcat` instead of `Catalina`. I tried to override the mbean from its configuration file _tomcat.yml_, however it doesn't seem possible to set e.g. `cache.mbean` or `requests.mbean` ourselves.

Reply:
Hey @dagwieers, thanks for pointing this out. Do you know on what cases tomcat uses the `Tomcat` domain and when the `Catalina` one? I think the tomcat module should support both cases. Could you please open a new issue to add support for the `Tomcat` domain?
In the meantime you can consider using the Jolokia module with a custom configuration: https://www.elastic.co/guide/en/beats/metricbeat/7.9/metricbeat-metricset-jolokia-jmx.html

If you are trying to override the mappings used by the tomcat module, I think but you should be using each metricset separatedly, and use the `jmx.mappings` setting in the configuration for the overrides. But I am not sure if this would work well. It'd be for example something like this for the cache metricset:
```
- module: tomcat
  metricsets: [cache]
  jmx.mappings:
    - mbean: 'Tomcat....'
        attributes:
          - attr: hitCount
            field: hit.total
          - attr: size
            field: size.total.kb
          - attr: maxSize
            field: size.max.kb
          - attr: lookupCount
            field: lookup.total
          - attr: ttl
            field: ttl.ms
```

Reply:
Btw, it seems there was an attempt to change this also in the example configuration for Tomcat in the Prometheus jmx exporter: https://github.com/prometheus/jmx_exporter/pull/92

Reply:
> Do you know on what cases tomcat uses the Tomcat domain and when the Catalina one? I think the tomcat module should support both cases. Could you please open a new issue to add support for the Tomcat domain?

I am afraid I am not that experienced with Tomcat to tell you. But a way to influence that (or detect that) would be useful. I will open a ticket.

> In the meantime you can consider using the Jolokia module with a custom configuration: https://www.elastic.co/guide/en/beats/metricbeat/7.9/metricbeat-metricset-jolokia-jmx.html

Yes, in the meantime I created my own config that uses jolokia and this basically removed the need to use the tomcat module altogether. Personally, I am unsure what the tomcat module brings to the table if it is just a standard listing for dashboards. We might as well go with just a tomcat.yml like your example. (I will include mine here for others)

Reply:
BTW another concern I had was that I could not find the tomcat module in de codebase and it took some time to find this PR (given I did not have a reference). Now I see it is under x-pack, but that is non-trivial for people to find. And unfortunately searching for tomcat does not help ;-)

Maybe it would be worthwhile to merge modules together in the code, but split them for packaging/distribution/documentation instead?